x-pack/plugins/security_solution/public/management/pages/policy/view/ingest_manager_integration/endpoint_policy_create_extension.tsx

```diff
@@ -5,45 +5,461 @@
+          updatedPolicy: {
+            inputs: [
+              {
+                _config: {
+                  ...newPolicy.inputs[0]._config,
+                  eventFilters: {
+                    interactiveSession: true,
+                  },
+                },
+              },
+            ],
+          },
+        });
+      }
+    }, [onChange, newPolicy, radioInteractiveSelected, radioComprehensiveSelected]);
+
+    // / Endpoint Radio Options (NGAV and EDRs)
+    const [radioEndpointOption, setRadioEndpointOption] = useState('NGAV');
+
+    const onChangeRadioEndpoint = useCallback(
+      (e) => {
+        setRadioEndpointOption(e.target.value);
+        if (e.target.value === 'NGAV') {
+          onChange({
+            isValid: true,
+            updatedPolicy: {
+              inputs: [
+                {
+                  _config: {
+                    ...newPolicy.inputs[0]._config,
+                    endpointConfig: {
+                      preset: 'NGAV',
+                    },
+                  },
+                },
+              ],
+            },
+          });
+        } else if (e.target.value === 'EDR Essential') {
+          onChange({
+            isValid: true,
+            updatedPolicy: {
+              inputs: [
+                {
+                  _config: {
+                    ...newPolicy.inputs[0]._config,
+                    endpointConfig: {
+                      preset: 'EDREssential',
+                    },
+                  },
+                },
+              ],
+            },
+          });
+        } else if (e.target.value === 'EDR Complete') {
+          onChange({
+            isValid: true,
+            updatedPolicy: {
+              inputs: [
+                {
+                  _config: {
+                    ...newPolicy.inputs[0]._config,
+                    endpointConfig: {
+                      preset: 'EDRComplete',
+                    },
+                  },
+                },
+              ],
+            },
+          });
+        }
+      },
+      [onChange, newPolicy]
+    );
+    ///
+
+    const [checkboxMalwareChecked, setCheckboxMalwareChecked] = useState(false);
+    const [checkboxRansomwareChecked, setCheckboxRansomwareChecked] = useState(false);
+
+    const onChangeMalwareCheckbox = useCallback(() => {
+      setCheckboxMalwareChecked(!checkboxMalwareChecked);
+
+      onChange({
+        isValid: true,
+        updatedPolicy: {
+          inputs: [
+            {
+              ...newPolicy.inputs[0],
+              _config: {
+                ...newPolicy.inputs[0]._config,
+                cloudConfig: {
+                  ...newPolicy.inputs[0]._config.cloudConfig,
+                  preventions: {
+                    ...newPolicy.inputs[0]._config.cloudConfig.preventions,
+                    malware: !checkboxMalwareChecked,
+                  },
+                },
+              },
+            },
+          ],
+        },
+      });
```

Comment:
This part can be transformed into a function that accepts the prevention key and can be reused by the other preventions onChange methods